src/main/java/gregtech/common/crafting/GTFluidCraftingIngredient.java

```diff
@@ -0,0 +1,37 @@
+package gregtech.common.crafting;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandlerItem;
+
+import javax.annotation.Nullable;
+
+public class GTFluidCraftingIngredient extends Ingredient {
+    private final FluidStack fluidStack;
+
+    GTFluidCraftingIngredient(ItemStack... stacks) {
+        super(stacks);
+        this.fluidStack = stacks[0].getCapability(CapabilityFluidHandler.FLUID_HANDLER_ITEM_CAPABILITY, null).drain(Integer.MAX_VALUE, false);
```

Comment:
Should we throw an IllegalArgumentException here when the first ItemStack does not have the capability?